R/utilities-plotting.R

```diff
@@ -683,9 +707,74 @@
 
 .updateSpecificSetting <- function(specificSetting, specificPlotConfiguration, generalPlotConfiguration) {
   if (!is.null(specificPlotConfiguration[[specificSetting]]) &
-    !is.null(generalPlotConfiguration[[specificSetting]])) {
+      !is.null(generalPlotConfiguration[[specificSetting]])) {
     if (generalPlotConfiguration[[specificSetting]] != specificPlotConfiguration[[specificSetting]]) {
       specificPlotConfiguration[[specificSetting]] <- generalPlotConfiguration[[specificSetting]]
     }
   }
 }
+
+#' Names of aggregation available for .extractAggregatedSimulatedData
+#'
+#' @keywords internal
+SimulatedDataAggregationMethods <-
+  ospsuite.utils::enum(c(
+    "quantiles",
+    "arithmetic",
+    "geometric")
+  )
+
+#' Quantile Range
+#'
+#' @param x numeric vector to compute quantile range from
+#' @param probs numeric vector of length 3 containing probabilities with values in [0,1].
+#' @param na.rm logical; if true, any NA and NaN's are removed from x before the quantiles are computed.
+#' @param ... further arguments passed to stats::quantile.
+#'
+#' @return numeric vector of length 3 representing the computed quantiles of x.
+#' @noRd
+.quantRange <- function(x, probs = c(0.05, 0.5, 0.95), na.rm = FALSE, ...){
```

Comment:
Just for my understanding... There is no `@keyword internal` in the roxygen code here, but the function is still not exported (which is good). How does this work?

Reply:
This is because of `@noRd`

`Keyword internal` means there will be a manual page for the function but it won't be in the documentation index. [https://roxygen2.r-lib.org/articles/rd.html#indexing ](https://roxygen2.r-lib.org/articles/rd.html#indexing)" `@keywords internal` is useful because it removes the function from the documentation index; it’s useful for functions aimed primarily at other developers, not typical users of the package."

`@NoRd` means there won't even be a manual page so the `roxygen2` doc of that function only lives in the R script.

Reply:
I mean that it also cannot be found with `ospsuite::.quantRange`

Reply:
There is no `@export` tag either ;)

Reply:
True :D